ballerina/common_utils.bal

```diff
@@ -83,6 +83,22 @@ isolated function getInvalidArgumentValueError(string listIndexOfError, string e
     }
 }
 
+isolated function getCycleRecursiveFragmentError(string spreadName, map<parser:FragmentNode> visitedSpreads)
+    returns ErrorDetail {
+    Location[] locations = [];
+    string[] keys = visitedSpreads.keys();
+    int listIndex = <int>keys.indexOf(spreadName);
+    foreach int i in listIndex ..< keys.length() {
+        parser:FragmentNode fragmentSpread = visitedSpreads.get(keys[i]);
```

Comment:
Can't we replace this by this? We can remove the line no. 88 too.
```suggestion
    foreach string key in visitedSpreads.keys() {
        parser:FragmentNode fragmentSpread = visitedSpreads.get(key);
```

Reply:
We don't need to iterate the entire array here. Maybe we can slice the array.

Reply:
Hmm. In that case, this is okay. I misread the code it seems.